src/Sentry/SentryCheckIn.cs

```diff
@@ -0,0 +1,87 @@
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+
+namespace Sentry;
+
+/// <summary>
+/// Checkin metadata.
+/// </summary>
+public interface ICheckin
+{
+    /// <summary>
+    /// Checkin ID
+    /// </summary>
+    SentryId Id { get; }
+
+    /// <summary>
+    /// The distinct slug of the monitor.
+    /// </summary>
+    string MonitorSlug { get; }
+
+    /// <summary>
+    /// The status of the Checkin
+    /// </summary>
+    CheckinStatus Status { get; }
+}
+
+/// <summary>
+/// The Checkin Status
+/// </summary>
+public enum CheckinStatus
+{
+    /// <summary>
+    /// The Checkin is in progress
+    /// </summary>
+    InProgress,
+
+    /// <summary>
+    /// The Checkin is Ok
+    /// </summary>
+    Ok,
+
+    /// <summary>
+    /// The Checkin errored
+    /// </summary>
+    Error
+}
+
+/// <summary>
+/// Sentry Checkin
+/// </summary>
+// https://develop.sentry.dev/sdk/check-ins/
+public class SentryCheckIn : ICheckin, ISentryJsonSerializable
```

Comment:
The casing isn't aligned between the class and the interface. In Java they capitalized: `CheckIn`: https://github.com/getsentry/sentry-java/blob/a537f8a45828b751272940034c4a64ee5f8fd278/sentry/src/main/java/io/sentry/CheckIn.java#L14

Should we align?